### PR TITLE
Added worker type constraint to lease collection

### PR DESCRIPTION
**Pull request by nikhilgirraj**
Added a unique constraint to the cosmos db collection for leases. The code to handle a Http Conflict response due to this change is already present.

### CODE REVIEW

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs [hunk 1]
```diff
@@ -38,27 +39,34 @@ public class CosmosDbLeaseStore : ILeaseStore
...
             };
 
             collectionDefinition.PartitionKey.Paths.Add("/leaseType");
 
+            var leaseConstraint = new UniqueKey
+            {
+                Paths = new Collection<string> { "/leaseType" }
+            };
+
+            collectionDefinition.UniqueKeyPolicy.UniqueKeys.Add(leaseConstraint);
```
Comment (chomick) [inline src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:56]: Are we going to have a single collection for election requests and winner lease or will they be separated? @faprovitolo ?

Comment (chomick) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:56]: @nikhilgirraj what will happen if this code is executed on an existing collection? Will it fail or silently ignore the constraint?

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs [hunk 2]
```diff
@@ -38,27 +39,34 @@ public class CosmosDbLeaseStore : ILeaseStore
...
+
             await _documentClient.CreateDocumentCollectionIfNotExistsAsync(
                 UriFactory.CreateDatabaseUri(_options.Value.Database),
                 collectionDefinition,
-                new RequestOptions {OfferThroughput = _options.Value.OfferThroughput}).ConfigureAwait(false);
+                new RequestOptions { OfferThroughput = _options.Value.OfferThroughput }).ConfigureAwait(false);
```
Comment (chomick) [inline src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:61]: Can you change your VS settings so it does not modify formatting if code is not changed?

**Review by chomick — APPROVED** [review]
Few questions but overall ok

**Review by tabman83 — APPROVED** [review]

File: src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs (hunk 1 above)
Comment (nikhilgirraj) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:56]: different collections for both purposes - a new collection for all election requests is part of the next PR

Comment (nikhilgirraj) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:56]: It silently ignores the fact that the new request also specifies a Unique Key constraint.

Comment (faprovitolo) [reply src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs:56]: yes there will be a separate collection for leaseRequests having a relatively short DefaultTimeToLive, e.g. 30 seconds

**Pull request merged**